ansible/playbooks/roles/prometheus/tasks/upgrade/main.yml

```
@@ -0,0 +1,122 @@
+---
+# General things needed 
+- name: Prometheus | Include new version default variables
+  include_vars:
+    file: defaults/main.yml
+    name: prometheus_defaults
+
+- name: Prometheus | Populate service facts
+  service_facts:
+  when: ansible_facts.service is undefined
+  no_log: true
+
+
+# Prometheus upgrade
+- name: Prometheus upgrade | Set prometheus_will_upgrade fact
+  set_fact:
+    prometheus_will_upgrade: false
+
+- name: Prometheus upgrade | Check binary existance
+  stat:
+    path: /usr/local/bin/prometheus
+  register: prometheus_stat
+
+- name: Prometheus upgrade | Start upgrade
+  when:
+    - ansible_facts.services["prometheus.service"] is defined
+    - prometheus_stat.stat.exists
+  block:
+    - name: Prometheus upgrade | Get version
+      command: prometheus --version
+      changed_when: false
+      check_mode: false
+      register: __prometheus_version
```

```suggestion
      register: _prometheus_version
```
In Ansible we use just single underscore 😉

---

Gotcha, fixed for all instances.